src/interactive-window/editor-integration/cellhashprovider.ts

```diff
@@ -1,559 +0,0 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-'use strict';
-import * as hashjs from 'hash.js';
-import {
-    Disposable,
-    NotebookCell,
-    NotebookCellKind,
-    Position,
-    Range,
-    TextDocumentChangeEvent,
-    TextDocumentContentChangeEvent,
-    Uri
-} from 'vscode';
-import * as localize from '../../platform/common/utils/localize';
-
-import { splitMultilineString } from '../../webviews/webview-side/common';
-import { IDebugService, IDocumentManager } from '../../platform/common/application/types';
-import { traceInfo, traceInfoIfCI } from '../../platform/logging';
-import { IConfigurationService } from '../../platform/common/types';
-import { IKernel } from '../../kernels/types';
-import { InteractiveWindowView } from '../../notebooks/constants';
-import { stripAnsi } from '../../platform/common/utils/regexp';
-import { getCellResource, uncommentMagicCommands } from './cellFactory';
-import { CellMatcher } from './cellMatcher';
-import { ICellHash, ICellHashProvider, ICellHashListener, IFileHashes } from './types';
-import { getAssociatedNotebookDocument } from '../../notebooks/controllers/kernelSelector';
-import { getInteractiveCellMetadata } from '../helpers';
-import { getDisplayPath, getFilePath } from '../../platform/common/platform/fs-paths';
-import { IPlatformService } from '../../platform/common/platform/types';
-import { untildify } from '../../platform/common/utils/platform';
-
-// eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
-const _escapeRegExp = require('lodash/escapeRegExp') as typeof import('lodash/escapeRegExp'); // NOSONAR
-const LineNumberMatchRegex = /(;32m[ ->]*?)(\d+)(.*)/g;
-
-interface IRangedCellHash extends ICellHash {
-    code: string;
-    startOffset: number;
-    endOffset: number;
-    deleted: boolean;
-    realCode: string;
-    trimmedRightCode: string;
-    firstNonBlankLineIndex: number; // zero based. First non blank line of the real code.
-}
-
-// This class provides hashes for debugging jupyter cells. Call getHashes just before starting debugging to compute all of the
-// hashes for cells.
-export class CellHashProvider implements ICellHashProvider {
```

Comment:
Now split into:
* Code generate
* Code storage provider
* Traceback formatter (two classes, one for nb & one for IW)